src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/AdditionalFilesItemHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+
+namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices.Handlers
+{
+    /// <summary>
+    ///     Handles changes to the  &lt;AdditionalFiles/&gt; item during design-time builds.
+    /// </summary>
+    [Export(typeof(ILanguageServiceCommandLineHandler))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasicLanguageService2)]
+    internal class AdditionalFilesItemHandler : ILanguageServiceCommandLineHandler
+    {
+        private IWorkspaceProjectContext _context;
+
+        [ImportingConstructor]
+        public AdditionalFilesItemHandler()
+        {
+        }
+
+        public void SetContext(IWorkspaceProjectContext context)
```

Comment:
>SetContext [](start = 20, length = 10)

This class is exported in global scope, but consumed from ILanguageServiceRuleHandler(s) that are in UnconfiguredProject scope. SetContext might override contexts when called for different projects and Handle might be never called for some projects?